### PR TITLE
fix(Popup): do not dismiss context Popup on nested Popup scrolling

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
The problem:
When a popup is opened from right click (`on={'context'}`), it is dismissed when the scrolling happens outside of it.
If the Popup has a nested Popup, scrolling inside the nested Popup will dismiss both Popups. This is incorrect.

Example:
https://codesandbox.io/s/fluent-ui-example-forked-dcq8r4?file=/example.js
Scrolling in the nested Popup content dismisses both Popups.

This PR fix the way used by Popup to determine if scrolls happens within its content. After this fix, scrolling within nested Popup does not dismiss both the parent and child Popup. Scrolling outside still works as before.

### CODE REVIEW

**Comment by fabricteam** [comment]
## Perf Analysis (`@fluentui/react-northstar`)
<details><summary>Perf tests with no regressions</summary>

Scenario | Current PR Ticks | Baseline Ticks | Ratio
:--- | ---:| ---:| ---:
BoxMinimalPerf.default | [413](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/BoxMinimalPerf.default.html) | [371](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/BoxMinimalPerf.default_base.html) | 1.11:1
PortalMinimalPerf.default | [209](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/PortalMinimalPerf.default.html) | [189](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/PortalMinimalPerf.default_base.html) | 1.11:1
SegmentMinimalPerf.default | [409](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/SegmentMinimalPerf.default.html) | [371](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/SegmentMinimalPerf.default_base.html) | 1.1:1
ChatDuplicateMessagesPerf.default | [343](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ChatDuplicateMessagesPerf.default.html) | [322](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ChatDuplicateMessagesPerf.default_base.html) | 1.07:1
DividerMinimalPerf.default | [434](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DividerMinimalPerf.default.html) | [404](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DividerMinimalPerf.default_base.html) | 1.07:1
HeaderMinimalPerf.default | [429](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/HeaderMinimalPerf.default.html) | [401](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/HeaderMinimalPerf.default_base.html) | 1.07:1
IconMinimalPerf.default | [701](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/IconMinimalPerf.default.html) | [656](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/IconMinimalPerf.default_base.html) | 1.07:1
ChatWithPopoverPerf.default | [477](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ChatWithPopoverPerf.default.html) | [449](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ChatWithPopoverPerf.default_base.html) | 1.06:1
RefMinimalPerf.default | [265](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/RefMinimalPerf.default.html) | [251](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/RefMinimalPerf.default_base.html) | 1.06:1
RadioGroupMinimalPerf.default | [519](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/RadioGroupMinimalPerf.default.html) | [495](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/RadioGroupMinimalPerf.default_base.html) | 1.05:1
StatusMinimalPerf.default | [789](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/StatusMinimalPerf.default.html) | [753](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/StatusMinimalPerf.default_base.html) | 1.05:1
TableMinimalPerf.default | [467](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TableMinimalPerf.default.html) | [446](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TableMinimalPerf.default_base.html) | 1.05:1
TextMinimalPerf.default | [403](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TextMinimalPerf.default.html) | [385](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TextMinimalPerf.default_base.html) | 1.05:1
VideoMinimalPerf.default | [754](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/VideoMinimalPerf.default.html) | [715](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/VideoMinimalPerf.default_base.html) | 1.05:1
AttachmentSlotsPerf.default | [1225](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AttachmentSlotsPerf.default.html) | [1180](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AttachmentSlotsPerf.default_base.html) | 1.04:1
FormMinimalPerf.default | [494](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/FormMinimalPerf.default.html) | [475](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/FormMinimalPerf.default_base.html) | 1.04:1
GridMinimalPerf.default | [392](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/GridMinimalPerf.default.html) | [377](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/GridMinimalPerf.default_base.html) | 1.04:1
InputMinimalPerf.default | [1427](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/InputMinimalPerf.default.html) | [1370](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/InputMinimalPerf.default_base.html) | 1.04:1
ListNestedPerf.default | [660](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ListNestedPerf.default.html) | [635](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ListNestedPerf.default_base.html) | 1.04:1
MenuMinimalPerf.default | [974](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/MenuMinimalPerf.default.html) | [940](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/MenuMinimalPerf.default_base.html) | 1.04:1
ProviderMergeThemesPerf.default | [1865](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ProviderMergeThemesPerf.default.html) | [1801](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ProviderMergeThemesPerf.default_base.html) | 1.04:1
ReactionMinimalPerf.default | [438](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ReactionMinimalPerf.default.html) | [422](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ReactionMinimalPerf.default_base.html) | 1.04:1
AnimationMinimalPerf.default | [605](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AnimationMinimalPerf.default.html) | [586](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AnimationMinimalPerf.default_base.html) | 1.03:1
DropdownManyItemsPerf.default | [804](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DropdownManyItemsPerf.default.html) | [782](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DropdownManyItemsPerf.default_base.html) | 1.03:1
ImageMinimalPerf.default | [429](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ImageMinimalPerf.default.html) | [418](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ImageMinimalPerf.default_base.html) | 1.03:1
ToolbarMinimalPerf.default | [1084](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ToolbarMinimalPerf.default.html) | [1055](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ToolbarMinimalPerf.default_base.html) | 1.03:1
CardMinimalPerf.default | [656](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/CardMinimalPerf.default.html) | [644](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/CardMinimalPerf.default_base.html) | 1.02:1
CheckboxMinimalPerf.default | [2918](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/CheckboxMinimalPerf.default.html) | [2860](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/CheckboxMinimalPerf.default_base.html) | 1.02:1
FlexMinimalPerf.default | [329](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/FlexMinimalPerf.default.html) | [322](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/FlexMinimalPerf.default_base.html) | 1.02:1
LayoutMinimalPerf.default | [419](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/LayoutMinimalPerf.default.html) | [411](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/LayoutMinimalPerf.default_base.html) | 1.02:1
SliderMinimalPerf.default | [1881](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/SliderMinimalPerf.default.html) | [1849](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/SliderMinimalPerf.default_base.html) | 1.02:1
TableManyItemsPerf.default | [2177](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TableManyItemsPerf.default.html) | [2137](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TableManyItemsPerf.default_base.html) | 1.02:1
TextAreaMinimalPerf.default | [583](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TextAreaMinimalPerf.default.html) | [572](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TextAreaMinimalPerf.default_base.html) | 1.02:1
TooltipMinimalPerf.default | [1151](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TooltipMinimalPerf.default.html) | [1126](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TooltipMinimalPerf.default_base.html) | 1.02:1
TreeMinimalPerf.default | [920](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TreeMinimalPerf.default.html) | [906](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TreeMinimalPerf.default_base.html) | 1.02:1
AvatarMinimalPerf.default | [227](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AvatarMinimalPerf.default.html) | [224](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AvatarMinimalPerf.default_base.html) | 1.01:1
ButtonOverridesMissPerf.default | [1874](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ButtonOverridesMissPerf.default.html) | [1849](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ButtonOverridesMissPerf.default_base.html) | 1.01:1
CarouselMinimalPerf.default | [537](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/CarouselMinimalPerf.default.html) | [530](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/CarouselMinimalPerf.default_base.html) | 1.01:1
ChatMinimalPerf.default | [845](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ChatMinimalPerf.default.html) | [839](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ChatMinimalPerf.default_base.html) | 1.01:1
DropdownMinimalPerf.default | [3236](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DropdownMinimalPerf.default.html) | [3209](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DropdownMinimalPerf.default_base.html) | 1.01:1
EmbedMinimalPerf.default | [4578](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/EmbedMinimalPerf.default.html) | [4548](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/EmbedMinimalPerf.default_base.html) | 1.01:1
HeaderSlotsPerf.default | [889](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/HeaderSlotsPerf.default.html) | [877](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/HeaderSlotsPerf.default_base.html) | 1.01:1
ItemLayoutMinimalPerf.default | [1339](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ItemLayoutMinimalPerf.default.html) | [1330](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ItemLayoutMinimalPerf.default_base.html) | 1.01:1
ListMinimalPerf.default | [587](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ListMinimalPerf.default.html) | [584](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ListMinimalPerf.default_base.html) | 1.01:1
LoaderMinimalPerf.default | [742](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/LoaderMinimalPerf.default.html) | [737](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/LoaderMinimalPerf.default_base.html) | 1.01:1
RosterPerf.default | [1247](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/RosterPerf.default.html) | [1238](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/RosterPerf.default_base.html) | 1.01:1
CustomToolbarPrototype.default | [4457](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/CustomToolbarPrototype.default.html) | [4420](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/CustomToolbarPrototype.default_base.html) | 1.01:1
AlertMinimalPerf.default | [316](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AlertMinimalPerf.default.html) | [317](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AlertMinimalPerf.default_base.html) | 1:1
ButtonSlotsPerf.default | [623](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ButtonSlotsPerf.default.html) | [621](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ButtonSlotsPerf.default_base.html) | 1:1
DatepickerMinimalPerf.default | [6034](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DatepickerMinimalPerf.default.html) | [6017](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DatepickerMinimalPerf.default_base.html) | 1:1
DialogMinimalPerf.default | [823](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DialogMinimalPerf.default.html) | [821](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/DialogMinimalPerf.default_base.html) | 1:1
LabelMinimalPerf.default | [446](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/LabelMinimalPerf.default.html) | [444](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/LabelMinimalPerf.default_base.html) | 1:1
SkeletonMinimalPerf.default | [402](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/SkeletonMinimalPerf.default.html) | [403](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/SkeletonMinimalPerf.default_base.html) | 1:1
SplitButtonMinimalPerf.default | [4782](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/SplitButtonMinimalPerf.default.html) | [4765](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/SplitButtonMinimalPerf.default_base.html) | 1:1
PopupMinimalPerf.default | [662](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/PopupMinimalPerf.default.html) | [671](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/PopupMinimalPerf.default_base.html) | 0.99:1
ProviderMinimalPerf.default | [1283](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ProviderMinimalPerf.default.html) | [1295](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ProviderMinimalPerf.default_base.html) | 0.99:1
AttachmentMinimalPerf.default | [181](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AttachmentMinimalPerf.default.html) | [185](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AttachmentMinimalPerf.default_base.html) | 0.98:1
ButtonMinimalPerf.default | [200](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ButtonMinimalPerf.default.html) | [204](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ButtonMinimalPerf.default_base.html) | 0.98:1
MenuButtonMinimalPerf.default | [1823](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/MenuButtonMinimalPerf.default.html) | [1855](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/MenuButtonMinimalPerf.default_base.html) | 0.98:1
TreeWith60ListItems.default | [208](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TreeWith60ListItems.default.html) | [213](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/TreeWith60ListItems.default_base.html) | 0.98:1
AccordionMinimalPerf.default | [177](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AccordionMinimalPerf.default.html) | [183](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/AccordionMinimalPerf.default_base.html) | 0.97:1
ListCommonPerf.default | [718](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ListCommonPerf.default.html) | [744](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ListCommonPerf.default_base.html) | 0.97:1
ListWith60ListItems.default | [697](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ListWith60ListItems.default.html) | [734](https://fluentuipr.z22.web.core.windows.net/pull/22087/perf-test-northstar/ListWith60ListItems.default_base.html) | 0.95:1

</details><div id="perfComment9424"/>

File: packages/fluentui/react-northstar/src/components/Popup/Popup.tsx [hunk 1]
```diff
@@ -497,7 +497,7 @@ export const Popup: React.FC<PopupProps> &
 
   const dismissOnScroll = (e: TouchEvent | WheelEvent) => {
     // we only need to dismiss if the scroll happens outside the popup
-    if (!popupContentRef.current.contains(e.target as Node)) {
+    if (!elementContains(popupContentRef.current, e.target as HTMLElement)) {
```
Comment (ling1726) [inline packages/fluentui/react-northstar/src/components/Popup/Popup.tsx:500]: nice, you figured out the virtual element all on your own ?

File: packages/fluentui/e2e/tests/popupDismissScroll.spec.ts [hunk 2]
```diff
@@ -25,4 +25,32 @@ describe('Popup - dismiss on scroll container', () => {
...
+      cy.visible(`#${selectors.nestedPopup.childPopupTriggerId}`); // popup 1 visible
+
+      cy.get(`#${selectors.nestedPopup.childPopupTriggerId}`).click(); // opens popup 2
+      cy.visible(`#${selectors.nestedPopup.childPopupContentId}`); // nested popup 2 visible
+    };
+
+    it('dismiss on scroll document', () => {
+      openNestedPopups();
+
+      cy.get('body').trigger('wheel'); // page scroll
```
Comment (ling1726) [inline packages/fluentui/e2e/tests/popupDismissScroll.spec.ts:41]: just being paranoid, did you check that this test fails without your change ?

**Review by ling1726 — APPROVED** [review]

File: packages/fluentui/e2e/tests/popupDismissScroll.spec.ts (hunk 2 above)
Comment (YuanboXue-Amber) [reply packages/fluentui/e2e/tests/popupDismissScroll.spec.ts:41]: Good reminder and yes I did :)

File: packages/fluentui/react-northstar/src/components/Popup/Popup.tsx (hunk 1 above)
Comment (YuanboXue-Amber) [reply packages/fluentui/react-northstar/src/components/Popup/Popup.tsx:500]: nope :D shift

**Pull request merged**